src/internal/ppsutil/util.go

```diff
@@ -435,16 +437,13 @@ func GetStatsCommit(commitInfo *pfs.CommitInfo) *pfs.Commit {
 // ContainsS3Inputs returns 'true' if 'in' is or contains any PFS inputs with
 // 'S3' set to true. Any pipelines with s3 inputs lj
 func ContainsS3Inputs(in *pps.Input) bool {
-	var found bool
-	pps.VisitInput(in, func(in *pps.Input) {
-		if found {
-			return
-		}
+	foundErr := pps.VisitInput(in, func(in *pps.Input) error {
 		if in.Pfs != nil && in.Pfs.S3 {
-			found = true
+			return errutil.ErrBreak
 		}
+		return nil
 	})
-	return found
+	return foundErr != nil
```

Comment:
Happy to revert this and the other cases that are looking for an input type if this is too "clever"/not what `ErrBreak` is expected to do.

Reply:
That's a good point, I'll put it back and make `ErrBreak` behave the way it does in other callback iterators